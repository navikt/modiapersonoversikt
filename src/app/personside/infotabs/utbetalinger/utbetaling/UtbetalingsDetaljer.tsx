import * as React from 'react';
import styled from 'styled-components';
import { Normaltekst } from 'nav-frontend-typografi';
import theme from '../../../../../styles/personOversiktTheme';
import {
    createTable,
    formaterNOK,
    skattBelopAscComparator,
    trekkBelopAscComparator,
    ytelseBelopDescComparator
} from '../utils/utbetalingerUtils';
import { Ytelse, Ytelseskomponent } from '../../../../../models/utbetalinger';
import { EnkeltYtelseProps } from './EnkeltYtelse';
import AlertStripeAdvarsel from 'nav-frontend-alertstriper/lib/advarsel-alertstripe';
import { Bold } from '../../../../../components/common-styled-components';
import DetaljerCollapse from '../DetaljerCollapse';

type Props = EnkeltYtelseProps;

const Wrapper = styled.aside`
  th:not(:first-child) {
    font-weight: normal;
    font-style: italic;
  }
`;

const OversiktStyle = styled.div`
  td:not(:first-child) {
    font-weight: bold;
  }
`;

const Border = styled.div`
  border-top: 2px solid ${theme.color.bakgrunn};
  margin: .5rem 0;
`;

function utbetalingsDetaljerTable(ytelse: Ytelse) {
    if (ytelse.ytelseskomponentListe || ytelse.trekkListe || ytelse.skattListe) {
        const tittelrekke = ['Detaljer', 'Sats', 'Antall', 'Beløp'];
        let tabellElementer: Array<Array<number | string | undefined>> = [];

        if (ytelse.ytelseskomponentListe) {
            ytelse.ytelseskomponentListe
                .sort(ytelseBelopDescComparator)
                .map((ytelseskomponent: Ytelseskomponent) =>
                    tabellElementer.push([
                        ytelseskomponent.ytelseskomponenttype,
                        ytelseskomponent.satsbeløp ? formaterNOK(ytelseskomponent.satsbeløp) : '',
                        ytelseskomponent.satsantall,
                        formaterNOK(ytelseskomponent.ytelseskomponentbeløp)
                    ]));
        }

        if (ytelse.skattListe) {
            ytelse.skattListe
                .sort(skattBelopAscComparator)
                .map(skattElement =>
                    tabellElementer.push([
                        'Skattetrekk',
                        '',
                        '',
                        formaterNOK(skattElement.skattebeløp)
                    ]));
        }

        if (ytelse.trekkListe) {
            ytelse.trekkListe
                .sort(trekkBelopAscComparator)
                .map(trekkElement =>
                    tabellElementer.push([
                        trekkElement.trekktype,
                        '',
                        '',
                        formaterNOK(trekkElement.trekkbeløp)
                    ]));
        }

        return createTable(
            tittelrekke,
            tabellElementer
        );
    } else {
        return <AlertStripeAdvarsel>Manglende data. Kunne ikke finne detaljer om utbetaling.</AlertStripeAdvarsel>;
    }
}

function UtbetalingsDetaljer(props: Props) {
    const ytelse = props.ytelse;
    const detaljer = utbetalingsDetaljerTable(ytelse);
    const oversikt = createTable(
        ['Konto', 'Brutto', 'Trekk', 'Utbetalt'],
        [[
            props.konto,
            formaterNOK(ytelse.ytelseskomponentersum),
            formaterNOK(ytelse.skattsum + ytelse.trekksum),
            formaterNOK(ytelse.nettobeløp)
        ]]
    );
    return (
        <DetaljerCollapse open={props.visDetaljer} toggle={props.toggleVisDetaljer}>
            <Wrapper>
                <Normaltekst tag="span">
                    <OversiktStyle>
                        {oversikt}
                    </OversiktStyle>
                </Normaltekst>
                <Border/>
                <Normaltekst tag="span">
                    {detaljer}
                </Normaltekst>
                <Border/>
                <Normaltekst tag={'h4'}><Bold>Melding</Bold></Normaltekst>
                <Normaltekst>{props.melding || ''}</Normaltekst>
            </Wrapper>
        </DetaljerCollapse>
    );
}

export default UtbetalingsDetaljer;
