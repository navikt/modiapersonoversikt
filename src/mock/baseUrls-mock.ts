import { BaseUrlsResponse } from '../models/baseurls';

export function mockBaseUrls(): BaseUrlsResponse {
    return {
        baseUrls: [
            {
                key: 'norg2-frontend',
                url: 'https://norg2-frontend.nais.preprod.local'
            }
        ]
    };
}
