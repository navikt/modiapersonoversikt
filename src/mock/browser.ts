import { setupWorker } from 'msw/browser';
import { handlers } from './';

export const worker = setupWorker(...handlers);
