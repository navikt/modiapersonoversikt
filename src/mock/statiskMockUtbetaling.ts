import { Utbetaling, Ytelse } from '../models/utbetalinger';

export const statiskMockYtelse: Ytelse = {
    type: 'Sykepenger',
    ytelseskomponentListe: [
        {
            ytelseskomponenttype: 'Mouse',
            satsbeløp: 1930,
            satstype: 'Satstype',
            satsantall: 2,
            ytelseskomponentbeløp: 3860
        },
        {
            ytelseskomponenttype: 'Computer',
            satsbeløp: 1310,
            satstype: 'Satstype',
            satsantall: 3,
            ytelseskomponentbeløp: 3930
        },
        {
            ytelseskomponenttype: 'Bacon',
            satsbeløp: 558,
            satstype: 'Satstype',
            satsantall: 10,
            ytelseskomponentbeløp: 5580
        }
    ],
    ytelseskomponentersum: 13370,
    trekkListe: [
        {
            trekktype: 'Prosenttrekk',
            trekkbeløp: -1042,
            kreditor: 'Bjerke, Berntsen and Nilsen'
        },
        {
            trekktype: 'Prosenttrekk',
            trekkbeløp: -212,
            kreditor: 'Olsen AS'
        }
    ],
    trekksum: -1254,
    skattListe: [
        {
            skattebeløp: -730
        }
    ],
    skattsum: -730,
    nettobeløp: 11386,
    periode: {
        start: '2016-11-09',
        slutt: '2017-10-31'
    },
    bilagsnummer: '1107807621',
    arbeidsgiver: {
        navn: 'Ensjø Bistro',
        orgnr: '1234556678235'
    }
};

export const statiskMockUtbetaling: Utbetaling = {
    utbetaltTil: 'AREMARK TESTFAMILIEN',
    erUtbetaltTilPerson: true,
    erUtbetaltTilOrganisasjon: false,
    erUtbetaltTilSamhandler: false,
    nettobeløp: 11386,
    posteringsdato: '2017-03-26',
    utbetalingsdato: '2017-03-29',
    forfallsdato: null,
    melding: 'Utbetalingsmelding',
    metode: 'Bankkontooverføring',
    status: 'Utbetalt',
    konto: '83980854563',
    ytelser: [statiskMockYtelse]
};
