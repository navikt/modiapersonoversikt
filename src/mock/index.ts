import faker from 'faker/locale/nb_NO';
import navfaker from 'nav-faker';
import Cookies from 'js-cookie';
import FetchMock, {
    HandlerArgument,
    Middleware,
    MiddlewareUtils,
    MockHandler,
    ResponseUtils
} from 'yet-another-fetch-mock';
import { erGyldigFødselsnummer } from 'nav-faker/dist/personidentifikator/helpers/fodselsnummer-utils';

import { apiBaseUri } from '../api/config';
import { getPerson } from './person/personMock';
import { getMockKontaktinformasjon } from './person/krrKontaktinformasjon/kontaktinformasjon-mock';
import { mockGeneratorMedEnhetId, mockGeneratorMedFødselsnummer, withDelayedResponse } from './utils/fetch-utils';
import { getMockNavKontor } from './navkontor-mock';
import { erEgenAnsatt } from './egenansatt-mock';
import { mockBaseUrls } from './baseUrls-mock';
import { getMockVeilederRoller } from './veilderRoller-mock';
import { mockRetningsnummereKodeverk } from './kodeverk/retningsnummer-mock';
import { mockTilrettelagtKommunikasjonKodeverk } from './kodeverk/tilrettelagt-kommunikasjon-kodeverk-mock';
import { mockPostnummere } from './kodeverk/postnummer-kodeverk-mock';
import { mockLandKodeverk } from './kodeverk/land-kodeverk-mock';
import { mockValutaKodeverk } from './kodeverk/valuta-kodeverk-mock';
import { mockVergemal } from './person/vergemal/vergemalMock';
import { getMockUtbetalinger } from './utbetalinger/utbetalinger-mock';
import { getMockSykepengerRespons } from './ytelse/sykepenger-mock';
import { getMockForeldrepenger } from './ytelse/foreldrepenger-mock';
import { getMockPleiepenger } from './ytelse/pleiepenger-mock';
import { mockFeatureToggle } from './featureToggle-mock';
import { getMockSaksoversikt } from './saksoversikt/saksoversikt-mock';
import { getMockOppfølging, getMockYtelserOgKontrakter } from './oppfolging-mock';
import { getDittNavVarsler, getMockVarsler } from './varsler/varsel-mock';
import { getMockSlaaSammen } from './meldinger/meldinger-mock';
import { getForeslattEnhet, getMockAnsatte, getMockEnheter, getMockGsakTema } from './meldinger/oppgave-mock';
import { getMockInnloggetSaksbehandler } from './innloggetSaksbehandler-mock';
import { gsakSaker, pesysSaker } from './journalforing/journalforing-mock';
import { mockPersonsokResponse, mockStaticPersonsokRequest } from './person/personsokMock';
import { setupWsControlAndMock } from './context-mock';
import standardTekster from './standardTeksterMock.js';
import { mockTilgangTilSlett } from './meldinger/merk-mock';
import { MeldingerBackendMock } from './mockBackend/meldingerBackendMock';
import { getSaksBehandlersEnheterMock } from './getSaksBehandlersEnheterMock';
import { saksbehandlerCookieNavnPrefix } from '../redux/session/saksbehandlersEnhetCookieUtils';
import { OppgaverBackendMock } from './mockBackend/oppgaverBackendMock';
import { setupSaksbehandlerInnstillingerMock } from './saksbehandlerinnstillinger-mock';
import { failurerateMiddleware } from './utils/failureMiddleware';
import { setupDraftMock } from './draft-mock';
import tilgangskontrollMock from './tilgangskontroll-mock';

const STATUS_OK = () => 200;
const STATUS_BAD_REQUEST = () => 400;

const oppgaveBackendMock = new OppgaverBackendMock();
const meldingerBackendMock = new MeldingerBackendMock(oppgaveBackendMock);

export function randomDelay() {
    if (navfaker.random.vektetSjanse(0.05)) {
        return faker.random.number(5000);
    }
    return faker.random.number(750);
}

const fødselsNummerErGyldigStatus = (args: HandlerArgument) =>
    erGyldigFødselsnummer(args.pathParams.fodselsnummer) ? STATUS_OK() : STATUS_BAD_REQUEST();

function setupInnloggetSaksbehandlerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/hode/me',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockInnloggetSaksbehandler())
    );
}

function setUpSaksbehandlersEnheterMock(mock: FetchMock) {
    mock.get(apiBaseUri + '/hode/enheter', withDelayedResponse(randomDelay(), STATUS_OK, getSaksBehandlersEnheterMock));
}

function setupTilgangskontroll(mock: FetchMock) {
    mock.get(
        '/modiapersonoversikt-api/rest/tilgang/:fodselsnummer?',
        withDelayedResponse(
            randomDelay(),
            () => (Math.random() > 0.98 ? 400 : 200),
            mockGeneratorMedFødselsnummer(tilgangskontrollMock)
        )
    );
}

function setupPersonMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fødselsnummer => getPerson(fødselsnummer))
        )
    );
}

function setupEgenAnsattMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/egenansatt/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fødselsnummer => erEgenAnsatt(fødselsnummer))
        )
    );
}

function setupKontaktinformasjonMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/kontaktinformasjon',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fødselsnummer => getMockKontaktinformasjon(fødselsnummer))
        )
    );
}

function setupSaksoversiktMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/saker/:fodselsnummer/sakstema',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(getMockSaksoversikt)
        )
    );
}

function setupUtbetalingerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/utbetaling/:fodselsnummer',
        withDelayedResponse(randomDelay(), fødselsNummerErGyldigStatus, args =>
            getMockUtbetalinger(args.pathParams.fodselsnummer, args.queryParams.startDato, args.queryParams.sluttDato)
        )
    );
}

function setupSykepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/sykepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockSykepengerRespons(fodselsnummer))
        )
    );
}

function setupForeldrepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/foreldrepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockForeldrepenger(fodselsnummer))
        )
    );
}

function setupPleiepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/pleiepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockPleiepenger(fodselsnummer))
        )
    );
}

function setupOppfølgingMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppfolging/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockOppfølging(fodselsnummer))
        )
    );
}

function setupYtelserOgKontrakter(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppfolging/:fodselsnummer/ytelserogkontrakter',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockYtelserOgKontrakter(fodselsnummer))
        )
    );
}

function setupVarselMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/varsler/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockVarsler(fodselsnummer))
        )
    );

    const dittnaveventHandler: MockHandler = ({ init }) => {
        const headers: any = init?.headers;
        const fnr = headers.fodselsnummer;
        if (!erGyldigFødselsnummer(fnr)) {
            return Promise.resolve({ status: 400 });
        }
        return Promise.resolve({ status: 200, body: JSON.stringify(getDittNavVarsler(fnr)) });
    };

    mock.get('/dittnav-eventer-modia/fetch/beskjed/all', ResponseUtils.delayed(randomDelay(), dittnaveventHandler));
    mock.get('/dittnav-eventer-modia/fetch/oppgave/all', ResponseUtils.delayed(randomDelay(), dittnaveventHandler));
}

function setupMeldingerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/dialog/:fodselsnummer/meldinger',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fodselsnummer => meldingerBackendMock.getMeldinger(fodselsnummer))
        )
    );
}

function setupSlaasammenMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fodselsnummer/slaasammen',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            mockGeneratorMedFødselsnummer(fodselsnummer => getMockSlaaSammen(fodselsnummer))
        )
    );
}

function setupGsakTemaMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/dialogoppgave/tema',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockGsakTema())
    );
}

function setupOppgaveEnhetMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/oppgavebehandlere/alle',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockEnheter())
    );
}

function setupForeslatteEnheterMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/oppgavebehandlere/foreslatte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getForeslattEnhet())
    );
}

function setupAnsattePaaEnhetMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/:enhetId/ansatte',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            mockGeneratorMedEnhetId(enhetId => getMockAnsatte(enhetId))
        )
    );
}

function setupTilgangTilSlettMock(mock: FetchMock) {
    mock.get(
        `${apiBaseUri}/dialogmerking/slett`,
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockTilgangTilSlett())
    );
}

function setupGeografiskTilknytningMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter',
        withDelayedResponse(
            randomDelay(),
            (args: HandlerArgument) => {
                if (isNaN(args.queryParams.gt) && !args.queryParams.dkode) {
                    return 404;
                } else {
                    return 200;
                }
            },
            (args: HandlerArgument) => getMockNavKontor(args.queryParams.gt, args.queryParams.dkode)
        )
    );
}

function setupPersonsokMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/personsok',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockPersonsokResponse(mockStaticPersonsokRequest()))
    );
}

function setSaksbehandlerinnstillingerMockBackend(args: HandlerArgument) {
    Cookies.set(saksbehandlerCookieNavnPrefix + '-Z990099', args.body);
    return args.body;
}

function setupVelgEnhetMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/hode/velgenhet',
        withDelayedResponse(randomDelay(), STATUS_OK, setSaksbehandlerinnstillingerMockBackend)
    );
}

function setupOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/oppgaver/plukk/:temagruppe',
        withDelayedResponse(randomDelay(), STATUS_OK, () => oppgaveBackendMock.plukkOppgave())
    );
}

function setupOpprettHenvendelseMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fnr/fortsett/opprett',
        withDelayedResponse(randomDelay(), STATUS_OK, request => meldingerBackendMock.opprettHenvendelse(request.body))
    );
}

function setupFerdigstillHenvendelseMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fnr/fortsett/ferdigstill',
        withDelayedResponse(randomDelay(), STATUS_OK, request => {
            meldingerBackendMock.ferdigstillHenvendelse(request.body);
            return {};
        })
    );
}

function setupSendDelsvarMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fnr/delvis-svar',
        withDelayedResponse(randomDelay(), STATUS_OK, request => {
            meldingerBackendMock.sendDelsvar(request.body);
            return {};
        })
    );
}

function setupTildelteOppgaverMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppgaver/tildelt',
        withDelayedResponse(randomDelay(), STATUS_OK, () => oppgaveBackendMock.getTildelteOppgaver())
    );
}

function setupLeggTilbakeOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/oppgaver/legg-tilbake',
        withDelayedResponse(randomDelay(), STATUS_OK, request => oppgaveBackendMock.leggTilbake(request.body))
    );
}

function setupSendReferatMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fodselsnummer/sendreferat',
        withDelayedResponse(randomDelay() * 2, STATUS_OK, request => {
            meldingerBackendMock.sendReferat(request.body);
            return {};
        })
    );
}

function setupSendSpørsmålMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fodselsnummer/sendsporsmal',
        withDelayedResponse(randomDelay() * 2, STATUS_OK, request => {
            meldingerBackendMock.sendSpørsmål(request.body);
            return {};
        })
    );
}

function setupSendSvarMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialog/:fodselsnummer/sendsvar',
        withDelayedResponse(randomDelay() * 2, STATUS_OK, () => {
            return {};
        })
    );
}

function setupVergemalMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/vergemal',
        withDelayedResponse(
            randomDelay(),
            fødselsNummerErGyldigStatus,
            mockGeneratorMedFødselsnummer(fødselsnummer => mockVergemal(fødselsnummer))
        )
    );
}

function setupBaseUrlsMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/baseurls',
        withDelayedResponse(randomDelay(), STATUS_OK, () => {
            return mockBaseUrls();
        })
    );
}

function setupFeatureToggleMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/featuretoggle/:toggleId',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            // @ts-ignore
            (args: HandlerArgument) => mockFeatureToggle(args.pathParams.toggleId)
        )
    );
}

function setupVeilederRollerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/veileder/roller',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockVeilederRoller())
    );
}

function setupRetningsnummerKodeverkMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Retningsnumre',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockRetningsnummereKodeverk())
    );
}

function setupTilrettelagtKommunikasjonKodeverkMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/TilrettelagtKommunikasjon',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockTilrettelagtKommunikasjonKodeverk())
    );
}

function setupPostnummerKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Postnummer',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockPostnummere())
    );
}

function setupLandKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Landkoder',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockLandKodeverk())
    );
}

function setupValutaKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Valutaer',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockValutaKodeverk())
    );
}

function setupJournalforingMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/journalforing/:fnr/saker/sammensatte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => gsakSaker)
    );
    mock.get(
        apiBaseUri + '/journalforing/:fnr/saker/pensjon',
        withDelayedResponse(randomDelay(), STATUS_OK, () => pesysSaker)
    );
    mock.post(
        apiBaseUri + '/journalforing/:fnr/:traadId',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function opprettOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogoppgave/opprett',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function merkAvsluttMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogmerking/avslutt',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function merkBidragMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogmerking/bidrag',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function merkFeilsendtMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogmerking/feilsendt',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function merkKontorsperretMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogmerking/kontorsperret',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function merkSlettMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogmerking/slett',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function setupStandardteksterMock(mock: FetchMock) {
    mock.get(
        '/modiapersonoversikt-skrivestotte/skrivestotte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => standardTekster)
    );

    mock.post(
        '/modiapersonoversikt-skrivestotte/skrivestotte/statistikk/:id',
        withDelayedResponse(randomDelay(), STATUS_OK, () => undefined)
    );
}

function setupAvsluttOppgaveGosysMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogmerking/avsluttgosysoppgave',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

const contentTypeMiddleware: Middleware = (requestArgs, response) => {
    if (response.headers) {
        return response;
    }
    response.headers = {
        'content-type': 'application/json'
    };
    return response;
};

console.log('=========================='); // tslint:disable-line
console.log('======== MED MOCK ========'); // tslint:disable-line
console.log('=========================='); // tslint:disable-line

const mock = FetchMock.configure({
    enableFallback: true,
    middleware: MiddlewareUtils.combine(
        contentTypeMiddleware,
        failurerateMiddleware(0.02),
        MiddlewareUtils.loggingMiddleware()
    )
});

setupInnloggetSaksbehandlerMock(mock);
setupPersonMock(mock);
setupTilgangskontroll(mock);
setupEgenAnsattMock(mock);
setupKontaktinformasjonMock(mock);
setupGeografiskTilknytningMock(mock);
setupSaksoversiktMock(mock);
setupUtbetalingerMock(mock);
setupSykepengerMock(mock);
setupForeldrepengerMock(mock);
setupPleiepengerMock(mock);
setupOppgaveMock(mock);
setupOpprettHenvendelseMock(mock);
setupFerdigstillHenvendelseMock(mock);
setupSendDelsvarMock(mock);
setupTildelteOppgaverMock(mock);
setupLeggTilbakeOppgaveMock(mock);
setupVergemalMock(mock);
setupBaseUrlsMock(mock);
setupFeatureToggleMock(mock);
setupVeilederRollerMock(mock);
setupRetningsnummerKodeverkMock(mock);
setupTilrettelagtKommunikasjonKodeverkMock(mock);
setupPostnummerKodeverk(mock);
setupWsControlAndMock(mock);
setupLandKodeverk(mock);
setupValutaKodeverk(mock);
setupOppfølgingMock(mock);
setupMeldingerMock(mock);
setupGsakTemaMock(mock);
setupOppgaveEnhetMock(mock);
setupForeslatteEnheterMock(mock);
setupAnsattePaaEnhetMock(mock);
setupTilgangTilSlettMock(mock);
setupYtelserOgKontrakter(mock);
setupVarselMock(mock);
opprettOppgaveMock(mock);
setupSendReferatMock(mock);
setupSendSpørsmålMock(mock);
setupSendSvarMock(mock);
setupPersonsokMock(mock);
merkAvsluttMock(mock);
merkBidragMock(mock);
merkFeilsendtMock(mock);
merkKontorsperretMock(mock);
merkSlettMock(mock);
setupJournalforingMock(mock);
setupStandardteksterMock(mock);
setupSlaasammenMock(mock);
setupVelgEnhetMock(mock);
setUpSaksbehandlersEnheterMock(mock);
setupAvsluttOppgaveGosysMock(mock);
setupSaksbehandlerInnstillingerMock(mock);
setupDraftMock(mock);
