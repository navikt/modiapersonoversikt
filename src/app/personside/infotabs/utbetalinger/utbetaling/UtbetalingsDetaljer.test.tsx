import * as React from 'react';
import * as renderer from 'react-test-renderer';
import UtbetalingsDetaljer from './UtbetalingsDetaljer';
import { statiskMockYtelse } from '../../../../../mock/statiskMockUtbetaling';

// Mock react collapse sin UnmountClosed
jest.mock('react-collapse', () => {
    return {
        // @ts-ignore
        UnmountClosed: props => props.children
    };
});

test('Viser utbetalingsdetaljer riktig med liste med ytelser og trekk', () => {
    const visittkortheader = renderer.create(
        <UtbetalingsDetaljer konto="123" melding="Dette er en melding" ytelse={statiskMockYtelse} />
    );

    const json = visittkortheader.toJSON();
    expect(json).toMatchSnapshot();
});
