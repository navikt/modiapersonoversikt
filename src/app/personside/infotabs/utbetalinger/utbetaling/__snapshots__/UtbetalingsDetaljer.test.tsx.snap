// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser utbetalingsdetaljer riktig med liste med ytelser og trekk 1`] = `
.c0 th:not(:first-child) {
  font-weight: normal;
  font-style: italic;
}

.c1 td:not(:first-child) {
  font-weight: 600;
}

.c2 {
  border-top: solid 0.0625rem #b7b1a9;
  margin: 0.5rem 0;
}

<aside
  aria-label="Utbetalingsdetaljer"
  className="c0"
>
  <span
    className="typo-normal"
  >
    <div
      className="c1"
    >
      <table
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              role="columnheader"
            >
              Konto
            </th>
            <th
              role="columnheader"
            >
              Brutto
            </th>
            <th
              role="columnheader"
            >
              Trekk
            </th>
            <th
              role="columnheader"
            >
              Utbetalt
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              123
            </td>
            <td
              role="cell"
            >
              13,370.00
            </td>
            <td
              role="cell"
            >
              -1,984.00
            </td>
            <td
              role="cell"
            >
              11,386.00
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </span>
  <div
    className="c2"
  />
  <span
    className="typo-normal"
  >
    <table
      role="table"
    >
      <thead
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="columnheader"
          >
            Detaljer
          </th>
          <th
            role="columnheader"
          >
            Sats
          </th>
          <th
            role="columnheader"
          >
            Antall
          </th>
          <th
            role="columnheader"
          >
            Beløp
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Bacon
          </td>
          <td
            role="cell"
          >
            558.00
          </td>
          <td
            role="cell"
          >
            10
          </td>
          <td
            role="cell"
          >
            5,580.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Computer
          </td>
          <td
            role="cell"
          >
            1,310.00
          </td>
          <td
            role="cell"
          >
            3
          </td>
          <td
            role="cell"
          >
            3,930.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Mouse
          </td>
          <td
            role="cell"
          >
            1,930.00
          </td>
          <td
            role="cell"
          >
            2
          </td>
          <td
            role="cell"
          >
            3,860.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Skattetrekk
          </td>
          <td
            role="cell"
          >
            —
          </td>
          <td
            role="cell"
          >
            —
          </td>
          <td
            role="cell"
          >
            -730.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Prosenttrekk
          </td>
          <td
            role="cell"
          >
            —
          </td>
          <td
            role="cell"
          >
            —
          </td>
          <td
            role="cell"
          >
            -1,042.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Prosenttrekk
          </td>
          <td
            role="cell"
          >
            —
          </td>
          <td
            role="cell"
          >
            —
          </td>
          <td
            role="cell"
          >
            -212.00
          </td>
        </tr>
      </tbody>
    </table>
  </span>
  <div
    className="c2"
  />
  <h4
    className="typo-element"
  >
    Melding
  </h4>
  <p
    className="typo-normal"
  >
    Dette er en melding
  </p>
</aside>
`;
